sklearn/metrics/ranking.py

```diff
@@ -292,7 +299,23 @@ def _binary_roc_auc_score(y_true, y_score, sample_weight=None):
 
         fpr, tpr, tresholds = roc_curve(y_true, y_score,
                                         sample_weight=sample_weight)
-        return auc(fpr, tpr)
+        if max_fpr is None or max_fpr == 1:
+            return auc(fpr, tpr)
+
+        idx = np.where(fpr <= max_fpr)[0]
+        # Linearly interpolate the ROC curve until max_fpr
```

Comment:
perhaps this should be clearer to say we're just adding a single point at max_fpr by interpolation